File: app/src/main/java/com/noisyninja/androidlistpoc/model/Gender.java
package com.noisyninja.androidlistpoc.model;

public enum Gender {
    ALL(0),
    MALE(1),
    FEMALE(2);

    private int value;

    private Gender(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public Gender next() {
        switch (value) {
            case 0:
                return MALE;
            case 1:
                return FEMALE;
            default:
                return ALL;
        }
    }
}

File: app/src/main/java/com/noisyninja/androidlistpoc/model/Gender.kt
package com.noisyninja.androidlistpoc.model

enum class Gender private constructor(val value: Int) {
    ALL(0),
    MALE(1),
    FEMALE(2);

    operator fun next(): Gender {
        when (value) {
            0 -> return MALE
            1 -> return FEMALE
            else -> return ALL
        }
    }
}
